### ISSUE TITLE
Tagging checks no longer work

### ISSUE THREAD

**Issue by lauriejones** [jameslnewell/publish-if-not-published#13]
Hey James!

I ran our `publish-preview` script yesterday and it appears the logic about only publishing packages with a version suffix when a tag is specified no longer works.

I accidentally published 5 packages with regular versions (e.g. `4.5.0`) to npm under the `preview` tag.

I'll try to look into what's going on over the next few days but for right now we're removing this `publish-preview` from our repo 😭

**Comment by jameslnewell** [comment]
👋 Hey @lauriejones!

Thanks for raising an issue! Can you please provide the command that you ran to reproduce the broken behaviour as I can't seem to find the issue after [adding a bunch of additional tests](https://github.com/jameslnewell/publish-if-not-published/commit/40d8d57e618b6b197f3f10a9de4e9da59059c4cb).

**Comment by lauriejones** [comment]
Nice! Well that is strange then.

We had this command in our monorepo:

```
"publish-preview": "lerna exec -- publish-if-not-published -- --registry https://registry.npmjs.org/ --tag preview"
```

Which when I ran against a codebase with a number of packages with new, unpublished versions, some with suffixes and some without, all were published.

You can see in this screenshot the moment I realised what was happening and `Ctrl + C`'d my way outta there 😁 

![image](https://user-images.githubusercontent.com/11398624/95827297-471ca780-0d7f-11eb-83d8-ea75cd52d894.png)

All but the last of the 5 published packages should have been skipped. All were published under the preview tag.

I haven't jumped in to try again as I've been too scared to publish something accidentally again, but I can try and use the `--dry-run` flag you mentioned in #1 and see if I can reproduce the issue. It could definitely be something to do with our setup.

**Comment by jameslnewell** [comment]
I think I found it. The `tag-check` value is `false` by default so you have to explicitly pass in `--tag-check` to get it to do the check you're expecting. I believe this issue has existed since [the feature was introduced](https://github.com/jameslnewell/publish-if-not-published/pull/4/files#r360192229) 😄

Working script:
```
"publish-preview": "lerna exec publish-if-not-published -- --tag-check -- --registry https://registry.npmjs.org/ --tag preview"
```

Released `2.1.0` with [a fix](https://github.com/jameslnewell/publish-if-not-published/commit/e52c7b07990279f67ecb0c734557fc5ea978abaa) for the documented expected behaviour.

Working script after fix:
```
"publish-preview": "lerna exec publish-if-not-published -- -- --registry https://registry.npmjs.org/ --tag preview"
```

**Comment by lauriejones** [comment]
Gah, who introduced this feature?! 😅 

Thanks for being so thorough in investigating it. Now I can bump up to `2.1.0` and my initial script will work as we hoped.

Perhaps I'll add in the `--tag-check` to be extra certain 😁
